dataservice.js

```diff
@@ -0,0 +1,47 @@
+
+
+const combinedProductMap = new Map();
+const productMap = new Map();
+const categoryMap = new Map();
+
+// function to load products JSON and convert it to a map.
+let getProducts = () => {
```

Comment:
```suggestion
const getProducts = () => {
```
Change it all methods accordinly